Tasks/AzureRmWebAppDeployment/Tests/L0JsonVarSub.ts

```diff
@@ -0,0 +1,228 @@
+import ma = require('vsts-task-lib/mock-answer');
+import tmrm = require('vsts-task-lib/mock-run');
+import path = require('path');
+
+let taskPath = path.join(__dirname, '..', 'azurermwebappdeployment.js');
+let tr: tmrm.TaskMockRunner = new tmrm.TaskMockRunner(taskPath);
+tr.setInput('ConnectedServiceName', 'AzureRMSpn');
+tr.setInput('WebAppName', 'mytestapp');
+tr.setInput('Package', 'webAppPkg.zip');
+tr.setInput('UseWebDeploy', 'true');
+tr.setInput('JSONVariableSubstitutionsFlag', 'true');
+tr.setInput('JSONVariableSubstitutions', '');
+
+process.env['TASK_TEST_TRACE'] = 1;
+process.env["ENDPOINT_AUTH_AzureRMSpn"] = "{\"parameters\":{\"serviceprincipalid\":\"spId\",\"serviceprincipalkey\":\"spKey\",\"tenantid\":\"tenant\"},\"scheme\":\"ServicePrincipal\"}";
+process.env["ENDPOINT_DATA_AzureRMSpn_SUBSCRIPTIONNAME"] = "sName";
+process.env["ENDPOINT_DATA_AzureRMSpn_SUBSCRIPTIONID"] =  "sId";
+process.env["AZURE_HTTP_USER_AGENT"] = "TFS_useragent";
+process.env["SYSTEM_DEFAULTWORKINGDIRECTORY"] =  "DefaultWorkingDirectory";
+process.env["BUILD_SOURCEVERSION"] = "46da24f35850f455185b9188b4742359b537076f";
+process.env["BUILD_BUILDID"] = 1,
+process.env["RELEASE_RELEASEID"] = 1;
+process.env["BUILD_BUILDNUMBER"] = 1;
+process.env["RELEASE_RELEASENAME"] = "Release-1";
+process.env["BUILD_REPOSITORY_PROVIDER"] = "TfsGit";
+process.env["BUILD_REPOSITORY_NAME"] = "MyFirstProject";
+process.env["SYSTEM_TEAMFOUNDATIONCOLLECTIONURI"] = "https://abc.visualstudio.com/";
+process.env["SYSTEM_TEAMPROJECT"] = "MyFirstProject";
+process.env["BUILD_SOURCEVERISONAUTHOR"] = "author";
+process.env["RELEASE_RELEASEURI"] = "vstfs:///ReleaseManagement/Release/1";
+process.env["AGENT_NAME"] = "author";
+
+// provide answers for task mock
+let a: ma.TaskLibAnswers = <ma.TaskLibAnswers>{
+    "which": {
+        "cmd": "cmd"
+    },
+    "stats": {
+    	"webAppPkg.zip": {
+    		"isFile": true
+    	}
+    },
+    "osType": {
+        "osType": "Windows"
+    },
+    "checkPath": {
+        "cmd": true
```

Comment:
since utiltiy.findFiles method is added you might have to add more answers for checkpath